### DISCUSSION TITLE
Berry https - unsupported protocol

### DISCUSSION THREAD

**Discussion by lacrimal** [arendst/Tasmota#14524] [Q&A] [answered]
Hi
Trying to execute https example from doc https://tasmota.github.io/docs/Berry/#webclient-class o my ESP32-D0WDQ5 10.1.0(TasmoCompiler-esp32generic) it gives error:
```
cl = webclient()
cl.begin("https://raw.githubusercontent.com/tasmota/autoconf/main/esp32/M5Stack_Fire_autoconf.zip")
BRY: Exception> 'value_error' - unsupported protocol
stack traceback:
	<native>: in native function
	input:2: in function `main`
```
Not encrypted http is working.
Tried with downloaded tasmota32.bin v10.1.0 and custom firmware with flags:
```
#define USE_WEBCLIENT
#define USE_WEBCLIENT_HTTPS
```
no luck.
Should I do (compile or execute in berry) something more to get webclient https working ?
Thanks

**Comment by sfromis** [comment, marked as answer]
Having https included as default is a pretty recent development. Your test case above gives no error on current dev 2022.01.3 precompiled  tasmota32.bin, but failed on release tasmota32.bin 10.1.0

**Reply by lacrimal** [reply]
Sorry , same time updated answer, below addded to user_config_override.h resolve issue.
```
#define USE_WEBCLIENT
#define USE_WEBCLIENT_HTTPS
#define USE_TLS
```
Compiled against v10.1.0

**Reply by sfromis** [reply]
Just forget about compiling for 10.1.0 when the current state is that https is included by default. Use the development branch.

**Reply by lacrimal** [reply]
Thought that v10.1.0 is stable release - is it better to add extra things to stable than going to development branch for only few extra features ? Also get it unsuccessful to compile against development but in matter of fac t it was 229 commits ago ...

**Reply by sfromis** [reply]
As I see it, dev releases are also "stable enough", albeit with a risk of new errors not fixed yet. In fact, I'm running all my stuff on fresh development builds, and the number of issues I run into is quite low. I'd not bother sticking to last release version unless having problems with the development branch version. Part of the reasoning is that while development may take a few tries for some change working, most builds for existing stuff are not hit. Obviously, I'm careful with weasel words like "most" ;-) 

Don't know what your compile issues are, but for some reason the PlatformIO setup is a bit brittle, most "surprising" compile problem are mitigated by a clean slate PlatformIO, meaning deleting hidden folders (which will be rebuilt automatically).

**Reply by Jason2866** [reply]
Most of the time using the development branch is a good idea. Bug fixes are done and new features are introduced which may not work always with first attempt.
Stable is not a special variant or build. It is a release of a development version from time to time.

**Reply by lacrimal** [reply]
Thanks for explanations. Switched to development, pulled changes and it compiled just fine. Without special flags https works as expected.
Think we can close this.
